src/modules/job-info/watch.c

```diff
@@ -468,14 +466,12 @@ void watch_cancel_cb (flux_t *h, flux_msg_handler_t *mh,
                       const flux_msg_t *msg, void *arg)
 {
     struct info_ctx *ctx = arg;
-    uint32_t matchtag;
-
-    if (flux_request_unpack (msg, NULL, "{s:i}", "matchtag", &matchtag) < 0) {
-        flux_log_error (h, "%s: flux_request_unpack", __FUNCTION__);
+    if (flux_request_decode (msg, NULL, NULL) < 0) {
```

Comment:
Checking that the cancel request is a request is not super useful, since that would have already been checked before dispatching the message to this callback, and we can't respond with EPROTO to it since it is a noresponse.

Reply:
I wasn't sure if it was necessary as well, but I noticed a `flux_request_decode()` in the disconnect callback.  Should it be removed there as well?  The pattern is in multiple modules (minimally `job-info`, `kvs`, and `kvs-watch`).